### PR TITLE
Controls and cleanup

### PR DESCRIPTION
**Pull request by cixzhang**
No logic changes, just a bit of cleanup.
- Separated controls from renderer logic
- Wrapping getContext in try catch and no longer ignoring that test suite
- Cleaned up testing so we no longer need bundle.js
- added tests for controls

### CODE REVIEW

**Comment by ptpham** [comment]
Oh man. So much code that I was somehow unaware of. XD

File: src/renderer.js [hunk 1]
```diff
@@ -113,106 +133,21 @@ Renderer.prototype.getUniforms = function () {
...
-
-  return this.uniforms;
-};
+  uniforms.u_near = 1;
+  uniforms.u_far = 100;
+  uniforms.u_view = m4.identity([]);
+  uniforms.u_camera = m4.invert([], uniforms.u_view);
+  uniforms.u_world = camera.computedMatrix;
+  uniforms.u_worldView = m4.multiply([], uniforms.u_view, uniforms.u_world);
+  uniforms.u_worldViewProjection = m4.multiply([], projection, uniforms.u_worldView);
```
Comment (ptpham) [inline src/renderer.js:142]: Man, livin' la vida loca here with the gl-matrix API. =P

Comment (ptpham) [inline src/renderer.js:142]: (All I mean is that relying on the indexing allocation or whatever it's called has some mildly off flavor. =P)

Comment (ptpham) [inline src/renderer.js:142]: Don't worry about changing it though. XD

File: src/controls.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+      camera.rotate(Date.now(), -dx, dy);
+      currentX = mev.offsetX;
+      currentY = mev.offsetY;
+    }
+
+    function onMouseUp () {
+      canvas.removeEventListener('mousemove', onMouseMove);
+      canvas.removeEventListener('mouseup', onMouseUp);
+    }
```
Comment (ptpham) [inline src/controls.js:37]: Just a question to haunt you at night: What happens if mouseup doesn't fire on the element? O__O

File: src/renderer.js (hunk 1 above)
Comment (cixzhang) [inline src/renderer.js:142]: I still don't know what you mean :P

File: src/controls.js (hunk 2 above)
Comment (cixzhang) [inline src/controls.js:37]: I suppose I could listen on window...what if the mouseup doesn't fire on the window? ._.

Comment (ptpham) [inline src/controls.js:37]: O__O I think it's generally a bit dicey to modify handlers. My preference in this situation is to store state on the component (i.e. the canvas/viewer in this case). Do you know about the data-\* attributes? https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Using_data_attributes They can be a cleaner way of arbitrarily extending an element with state.

File: src/renderer.js (hunk 1 above)
Comment (ptpham) [inline src/renderer.js:142]: Oh, I just meant the automatic increase in the array length when something is assigned to an index. XD

**Pull request merged**